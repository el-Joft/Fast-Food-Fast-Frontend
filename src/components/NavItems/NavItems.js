import React from 'react';
import { Link } from 'react-router-dom';
import './NavItems.scss';

const NavItem = () => {
  const items = [
    {
      text: 'Home',
      link: '/',
    },
    {
      text: 'About Us',
      link: '/about',
    },
    {
      text: 'Login',
      link: '/login',
    },
    {
      text: 'Sign Up',
      link: '/signup',
    },
  ];

  const showItems = () => items.map((item, i) => (
    <li key={i}>
      <Link to={item.link}>
        {item.text}
      </Link>
    </li>
  ));

  return (
    <div>
      <nav>
        <ul className="nav-bar">
          {showItems()}
          <li className="cart-section">
            <a href="checkout.html">
              <img src="/public/images/icons/cart.png" alt="" />
              <span>
Cart(
                <span id="cartSystem" />
)
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  );
};

export default NavItem;
