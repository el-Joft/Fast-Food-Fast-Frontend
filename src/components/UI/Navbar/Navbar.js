/* eslint-disable import/no-named-as-default */
import React from 'react';
import './Navbar.scss';
import NavItem from '../../NavItems/NavItems';

export const Navbar = () => (
  <div>
    <NavItem />
  </div>
);

export default Navbar;
