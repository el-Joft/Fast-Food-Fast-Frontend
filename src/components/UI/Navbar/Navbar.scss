// /* NAV BAR */
// nav {
//   background: #42ce9f;
//   width: 90%;
//   border-radius: 5%;
//   display: flex;
//   position: relative;
// }

// .nav-bar {
//   display: flex;
//   padding: 20px;
//   justify-content: space-between;
//   li {
//     font-size: 1.5em;
//     margin-top: 10px;
//   }
//   a {
//     color: #fff;
//     padding-right: 40px;
//     font-family: Handlee;
//   }
//   img {
//     width: 30px;
//     height: 30px;
//     color: white;
//   }
// }

// .cart-section {
//   a {
//     display: flex;
//     margin-left: 50px;
//     padding-left: 20px;
//     font-family: Righteous;
//   }
//   span {
//     margin-left: 10px;
//   }
// }