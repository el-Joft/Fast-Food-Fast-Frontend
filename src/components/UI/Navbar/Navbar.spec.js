import React from 'react';
import { shallow } from 'enzyme';
import { NavItem } from '../../NavItems/NavItems';
import { Navbar } from './Navbar';

describe('shallow testing', () => {
  let wrapper;
  it('should test the component if it renders anything', () => {
    wrapper = shallow(<Navbar />);
    expect(wrapper.length).toEqual(1);
  });

  it('should test the component if it renders anything', () => {
    wrapper = shallow(<Navbar />);
    const li = wrapper.find('div');
    expect(li.length).toEqual(1);
  });

  it('renders One Comment Card components', () => {
    wrapper = shallow(<Navbar />);
    expect(wrapper.find(NavItem)).toEqual({});
  });

  it('should test the component if it renders anything', () => {
    wrapper = shallow(<Navbar />);
    expect(wrapper).toMatchSnapshot();
  });
});
