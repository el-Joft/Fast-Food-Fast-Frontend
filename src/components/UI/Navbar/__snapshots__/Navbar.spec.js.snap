// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow testing should test the component if it renders anything 1`] = `
<div>
  <Connect(NavItem) />
</div>
`;
